main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <random>
+#include <regex>
+
+void roll(int numberDice, int diceSides);
+void parseString(char *argument);
+
+int main(int argc, char *argv[]){
+//TODO Add some basic error checking
+    if(argc <= 1){
+        std::cout << "You haven't entered any arguments" << std::endl;
+        return -1;
+    }
+
+    for(int index = 1; index < argc; index++)
+         parseString(argv[index]);
+    return 0;
+}
+
+//Parses input then calls roll() on the arguments
+void parseString(char *argument){
+    //TODO Need to refactor, so unit testing is a viable strategy, possibly into class?
+    std::cmatch searchResults;
+
+    //Match any numbers at the begging of the line
+    std::regex firstNumberSearchString("^\\d+");
+
+    //Match any numbers at the end of line
+    std::regex secondNumberSearchString("\\d+$");
+
+    //Perform first search
+    regex_search(argument, searchResults, firstNumberSearchString);
+    int firstNumber = stoi(searchResults[0]);
+
+    //Perform second search
+    regex_search(argument, searchResults, secondNumberSearchString);
+    int secondNumber = stoi(searchResults[0]);
+
+    roll(firstNumber, secondNumber);
+}
+
+void roll(int numberDice = 0, int diceSides = 0){
+    //TODO Need to refactor, so unit testing is a viable strategy, possibly into class?
+    //Setup random number generator
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(1, diceSides);
+    
+    int sum {0};
```

Comment:
When you're doing a sum, it's best to think and make sure that "int" is big enough to actually hold the sum. In this case you're fine, but in other cases, you might need "long" or "long long".